releasenotes/notes/gomemlimit.yaml

```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: feature
+area: installation
+issue:
+- 40676
+
+releaseNotes:
+- |
+  **Added** an automatically set GOMEMLIMIT to `istiod` to reduce the risk of out-of-memory issues.
```

Comment:
nit: I was a little confused the first time I read this thinking the initial `an` should maybe be `and`. But after reading it a few times, `an` does make sense.  Could we maybe remove the `an automatically set` and not lose anything? or maybe Added GOMEMLIMIT, set to limits.memory, to `istiod` to reduce...